test/bus.test.js

```diff
@@ -0,0 +1,613 @@
+const chai = require('chai');
+const chaiHTTP = require('chai-http');
+const { app } = require('../app');
+const { expect } = require('chai');
+const buses = require('../models/buses.js');
+const db = require('../models');
+chai.should();
+chai.use(chaiHTTP);
+
+let token;
+describe('Crud operation for buses ', () => {
+	before((done) => {
+		db.users.destroy({
+			truncate: true,
+			restartIdentity: true,
+		});
+		done();
+	});
+	beforeEach(async () => {
+		const user = {
+			fullname: 'cyifuzo jean damascene',
+			username: 'cyifuzo',
+			email: 'operator@andela.com',
+			role: 'client',
+			password: 'test123',
+		};
+		await chai.request(app).post(`/api/v1/users/login/register`).send(user);
+	});
+	beforeEach(async () => {
+		let user = {
+			email: 'operator@andela.com',
+			password: 'test123',
+		};
+
+		const response = await chai
+			.request(app)
+			.post('/api/v1/users/login')
+			.send(user);
+
+		token = response.body.data.token;
+	});
+	afterEach(async () => {
+		await db.users.destroy({
+			truncate: true,
+			restartIdentity: true,
+		});
+	});
+
+	it('should Retieve all permissions', async () => {
+		const response = await chai.request(app).get(`/api/v1/buses`);
+		expect(response).to.have.status(200);
+	});
```

Comment:
We need to test the whole response object, not just the status code because if the method just returns the right code and not the right object keys and data then the test would be lying.

Reply:
Also, how are we expecting `/api/v1/buses` to return a list of all permissions? What am I not getting here? Please double-check or explain.